### PR TITLE
HDFS with framework authn

### PR DESCRIPTION
**Pull request by nickbp**

### CODE REVIEW

File: repo/packages/H/hdfs/5/config.json [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+					"id": "http://hdfs/dcos/mesosphere.com/service/placement_strategy",
+					"type": "string",
+					"description": "The placement strategy for HDFS nodes. This should always be NODE.",
+					"default": "NODE"
+				},
+				"secret_name": {
+					"type": "string",
+					"description": "Name of the authentication Secret to use for DC/OS service authentication. This should be left empty unless service authentication is needed.",
+					"default": ""
+				}
```
Comment (adam-mesos) [inline repo/packages/H/hdfs/5/config.json]: Any reason you didn't put this right next to principal?

File: repo/packages/H/hdfs/5/config.json [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+				},
+				"placement_strategy": {
+					"id": "http://hdfs/dcos/mesosphere.com/service/placement_strategy",
+					"type": "string",
+					"description": "The placement strategy for HDFS nodes. This should always be NODE.",
+					"default": "NODE"
+				},
+				"secret_name": {
+					"type": "string",
+					"description": "Name of the authentication Secret to use for DC/OS service authentication. This should be left empty unless service authentication is needed.",
```
Comment (adam-mesos) [inline repo/packages/H/hdfs/5/config.json]: Should probably mention that this is the name of the secret in the secret store, not a local file or some other name

File: repo/packages/H/hdfs/5/marathon.json.mustache [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+"env":{
+{{#service.secret_name}}
+"DCOS_SERVICE_ACCOUNT_CREDENTIAL": { "secret": "serviceCredential" },
+"MESOS_MODULES": "file:///opt/mesosphere/etc/mesos-scheduler-modules/dcos_authenticatee_module.json",
+"MESOS_AUTHENTICATEE": "com_mesosphere_dcos_ClassicRPCAuthenticatee",
+"SSL_CERT_FILE": "/run/dcos/pki/tls/certs/scheduler.crt",
+"SSL_KEY_FILE": "/run/dcos/pki/tls/private/scheduler.key",
+"SSL_CA_FILE": "/run/dcos/pki/CA/certs/ca.crt",
+"SSL_ENABLED": "true",
+{{/service.secret_name}}
```
Comment (adam-mesos) [inline repo/packages/H/hdfs/5/marathon.json.mustache]: So you're still bundling scheduler SSL in with authentication, which is fine for Argus, but for other customers, we may want to be able to enable SSL without authentication or v.v. See the Spark PR https://github.com/mesosphere/universe/pull/524

Comment (nickbp) [inline repo/packages/H/hdfs/5/marathon.json.mustache]: We can add more supported permutations once there's a customer asking for it

File: repo/packages/H/hdfs/5/config.json (hunk 1 above)
Comment (nickbp) [inline repo/packages/H/hdfs/5/config.json]: Expected usage

File: repo/packages/H/hdfs/5/config.json (hunk 2 above)
Comment (nickbp) [inline repo/packages/H/hdfs/5/config.json]: Done

**Comment by nickbp** [comment]
Closing this PR in favor of combining it with PR#524

**Pull request closed without merging**